capirca/lib/arista_tp.py

```diff
@@ -0,0 +1,1036 @@
+                                                           self.term, flags)
+
+            # tcp/udp port generation
+            port_str = self._processPorts(self.term)
+            if port_str != "":
+                protocol_str += port_str
+
+            # icmp[v6] handling
+            icmp_type_str = ""
+            icmp_code_str = ""
+            if self.term.protocol == ["icmp"] or \
+               self.term.protocol == ["icmpv6"]:
+                icmp_type_str, icmp_code_str = self._processICMP(self.term)
+
+            if self.term.icmp_type:
+                protocol_str += icmp_type_str
+            if self.term.icmp_code:
+                protocol_str += icmp_code_str
+
+            # don't render emppty protocol strings.
+            if protocol_str != "":
+                term_block.append([MATCH_INDENT, protocol_str, False])
+
+            # ADDITIONAL SUPPORTED MATCH OPTIONS ------------------------------
+            # packet length
+            if self.term.packet_length:
+                term_block.append([MATCH_INDENT,
+                                   "ip length %s" % self.term.packet_length,
+                                   False])
+
+            # fragment offset
+            if self.term.fragment_offset:
+                term_block.append([
+                    MATCH_INDENT,
+                    "fragment offset %s" % self.term.fragment_offset,
+                    False
+                ])
+
+            if self.term.hop_limit:
+                term_block.append([
+                    MATCH_INDENT,
+                    "ttl %s" % self.term.hop_limit,
+                    False
+                ])
+
+            if self.term.ttl:
+                term_block.append([
+                    MATCH_INDENT,
+                    "ttl %s" % self.term.ttl,
+                    False
+                ])
+
+            if len(misc_options) > 0:
+                for mopt in misc_options:
+                    term_block.append([MATCH_INDENT, mopt, False])
+
+        # ACTION HANDLING
+        # if there's no action, then this is an implicit permit
+        current_action = self._ACTIONS.get(self.term.action[0])
+        # non-permit/drop actions should be added here
+        has_extra_actions = (
+            self.term.logging or
+            self.term.counter or
+            self.term.dscp_set
+        )
+
+        # if !accept - generate an action statement
+        # if accept and there are extra actions generate an actions statement
+        # if accept and no extra actions don't generate an actions statement
+        if self.term.action != ["accept"]:
+            term_block.append([MATCH_INDENT, "actions", False])
+            term_block.append([ACTION_INDENT, "%s" % current_action, False])
+        elif self.term.action == ["accept"] and has_extra_actions:
+            term_block.append([MATCH_INDENT, "actions", False])
+
+        if has_extra_actions:
+            # logging
```

Comment:
Could you do the same for 'syslog' as well, according to the traffic-policy config the log gets logged in the syslog :) 

Actually, to meet this requirement [0], it'd be best to add all positive logging actions here, and enable logging for them. So for _LOGGING = set(('true', 'True', 'syslog', 'local', 'log-both')), set logging. Only for 'disable', do nothing. 

[0] - https://github.com/google/capirca/blob/master/doc/generator_patterns.md#logging-is-supported-correctly-for-different-types-of-logging